handler.go

```diff
@@ -1495,3 +1498,134 @@ func errorString(err error) string {
+		return
+	}
+	def.Name = inputDefName
+
+	// Validate columnLabel & duplicate primaryKey
+	numPrimaryKey := 0
+	for _, field := range def.Fields {
+		if field.PrimaryKey {
+			numPrimaryKey += 1
+			if field.Name == index.columnLabel {
+				continue
+			} else {
+				err = fmt.Errorf("primary field's name not match columnLabel")
+				http.Error(w, err.Error(), http.StatusBadRequest)
+				return
+			}
+		}
+	}
+	if numPrimaryKey > 1 {
+		err = errors.New("duplicate primaryKey with other field")
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Create InputDefinition.
+	_, err = index.CreateInputDefinition(def)
+	if err == ErrInputDefinitionExists {
+		http.Error(w, err.Error(), http.StatusConflict)
+		return
+	} else if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	err = h.Broadcaster.SendSync(
+		&internal.CreateInputDefinitionMessage{
+			Index:      indexName,
+			Name:       inputDefName,
+			Definition: def,
+		})
+	if err != nil {
+		h.logger().Printf("problem sending CreateInputDefinition message: %s", err)
+	}
+
+	if err := json.NewEncoder(w).Encode(postInputDefinitionResponse{}); err != nil {
+		h.logger().Printf("response encoding error: %s", err)
+	}
+}
+
+// handleGetInputDefinition handles GET /input-definition request.
+func (h *Handler) handleGetInputDefinition(w http.ResponseWriter, r *http.Request) {
+	indexName := mux.Vars(r)["index"]
+	inputDefName := mux.Vars(r)["input-definition"]
+
+	//Find index.
+	index := h.Holder.Index(indexName)
+	if index == nil {
+		if err := json.NewEncoder(w).Encode(deleteIndexResponse{}); err != nil {
+			h.logger().Printf("response encoding error: %s", err)
+		}
+		return
+	}
+	inputDef, _ := index.inputDefinitions[inputDefName]
+	if err := json.NewEncoder(w).Encode(InputDefinitionInfo{
+		Frames: inputDef.frames,
+		Fields: inputDef.fields,
+	}); err != nil {
+		h.logger().Printf("write status response error: %s", err)
+	}
+
+}
+
+// handleDeleteInputDefinition handles DELETE /input-definition request.
+func (h *Handler) handleDeleteInputDefinition(w http.ResponseWriter, r *http.Request) {
+	indexName := mux.Vars(r)["index"]
+	inputDefName := mux.Vars(r)["input-definition"]
+
+	// Find index.
+	index := h.Holder.Index(indexName)
+	if index == nil {
+		if err := json.NewEncoder(w).Encode(deleteIndexResponse{}); err != nil {
+			h.logger().Printf("response encoding error: %s", err)
+		}
+		return
+	}
+
+	// Delete input definition from the index.
+	if err := index.DeleteInputDefinition(inputDefName); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	err := h.Broadcaster.SendSync(
+		&internal.DeleteInputDefinitionMessage{
+			Index: indexName,
+			Name:  inputDefName,
+		})
+	if err != nil {
+		h.logger().Printf("problem sending CreateInputDefinition message: %s", err)
+	}
+
+	if err := json.NewEncoder(w).Encode(postInputDefinitionResponse{}); err != nil {
```

Comment:
since they are both empty struct, I will change it to `defaultInputDefinitionResponse` if it sounds ok